mealpy/mealpy.py

```diff
@@ -204,19 +207,47 @@ def cli_list():  # pragma: no cover
 
 @cli_list.command('cities', short_help='List available cities.')
 def cli_list_cities():  # pragma: no cover
-    cities = [i['name'] for i in MealPal.get_cities()]
-    print('\n'.join(cities))
+    print('\n'.join(list_cities()))
+
+
+def list_cities():
+    cities_file = CACHE_HOME / 'cities.json'
+    cities = [city['name'] for city in cache_list(cities_file, MealPal.get_cities)]
+
+    return cities
 
 
 @cli_list.command('restaurants', short_help='List available restaurants.')
 @click.argument('city')
 def cli_list_restaurants(city):  # pragma: no cover
-    restaurants = [i['restaurant']['name'] for i in MealPal.get_schedules(city)]
+    restaurants = [i['restaurant']['name'] for i in list_menu(city)]
     print('\n'.join(restaurants))
 
 
 @cli_list.command('meals', short_help='List meal choices.')
 @click.argument('city')
 def cli_list_meals(city):  # pragma: no cover
-    restaurants = [i['meal']['name'] for i in MealPal.get_schedules(city)]
+    restaurants = [i['meal']['name'] for i in list_menu(city)]
     print('\n'.join(restaurants))
+
+
+def list_menu(city):  # pragma: no cover
+    menu_file = CACHE_HOME / 'menu.json'
+    callback = partial(MealPal.get_schedules, city)
+    return cache_list(menu_file, callback)
```

Comment:
Should there be menu files per city? One might be exploring menus in multiple cities and retrieve the same menu in the cache.

Reply:
Good catch. I'll nest the results into a dictionary per city, instead of separate files..